File: course-seeder-bot/src/accountCounter.js
import fs from 'node:fs'
import path from 'node:path'
import { fileURLToPath } from 'node:url'

// Anchored to the project root regardless of the caller's cwd, so `npm start` from anywhere
// inside this package always reads/writes the same file. Gitignored on purpose: every clone (or
// every machine) starts its own numbering, there is nothing to share here.
const ROOT_DIR = path.dirname(path.dirname(fileURLToPath(import.meta.url)))
const COUNTER_FILE = path.join(ROOT_DIR, 'bot-account-counter.txt')

export function readCounter(defaultValue = 1) {
  try {
    const raw = fs.readFileSync(COUNTER_FILE, 'utf8').trim()
    const value = Number.parseInt(raw, 10)
    return Number.isInteger(value) && value > 0 ? value : defaultValue
  } catch {
    return defaultValue
  }
}

export function writeCounter(value) {
  fs.writeFileSync(COUNTER_FILE, String(value), 'utf8')
}

File: course-seeder-bot/seeder/account_counter.py
"""
Persists the next bot account number to try, next to this package. Gitignored on purpose: every
clone (or every machine) starts its own numbering, there is nothing to share here.
"""
from __future__ import annotations

from pathlib import Path

ROOT_DIR = Path(__file__).resolve().parent.parent
COUNTER_FILE = ROOT_DIR / "bot-account-counter.txt"


def read_counter(default_value: int = 1) -> int:
    try:
        raw = COUNTER_FILE.read_text(encoding="utf8").strip()
        value = int(raw)
        return value if value > 0 else default_value
    except (OSError, ValueError):
        return default_value


def write_counter(value: int) -> None:
    COUNTER_FILE.write_text(str(value), encoding="utf8")
